tests/components/enphase_envoy/test_select.py

```diff
@@ -71,106 +68,95 @@ async def test_no_select(
+        ("NC3", ("generator_action", "apply", "generator_action")),
+        ("NC3", ("microgrid_action", "apply", "micro_grid_action")),
+        ("NC3", ("grid_action", "shed", "grid_action")),
+    ],
+)
+@pytest.mark.parametrize(("action"), ["powered", "not_powered", "schedule", "none"])
 async def test_select_relay_actions(
     hass: HomeAssistant,
     mock_envoy: AsyncMock,
     config_entry: MockConfigEntry,
+    target: str,
+    relay: str,
+    action: str,
 ) -> None:
     """Test select platform entities dry contact relay actions."""
     with patch("homeassistant.components.enphase_envoy.PLATFORMS", [Platform.SELECT]):
         await setup_integration(hass, config_entry)
 
     entity_base = f"{Platform.SELECT}."
 
-    for contact_id, dry_contact in mock_envoy.data.dry_contact_settings.items():
-        name = dry_contact.load_name.lower().replace(" ", "_")
-        for target in (
-            ("generator_action", dry_contact.generator_action, "generator_action"),
-            ("microgrid_action", dry_contact.micro_grid_action, "micro_grid_action"),
-            ("grid_action", dry_contact.grid_action, "grid_action"),
-        ):
-            test_entity = f"{entity_base}{name}_{target[0]}"
-            assert (entity_state := hass.states.get(test_entity))
-            assert RELAY_ACTION_MAP[target[1]] == (current_state := entity_state.state)
-            # set all relay modes except current mode
-            for action in [action for action in ACTION_OPTIONS if not current_state]:
-                await hass.services.async_call(
-                    SELECT_DOMAIN,
-                    SERVICE_SELECT_OPTION,
-                    {
-                        ATTR_ENTITY_ID: test_entity,
-                        ATTR_OPTION: action,
-                    },
-                    blocking=True,
-                )
-                mock_envoy.update_dry_contact.assert_called_once_with(
-                    {"id": contact_id, target[2]: REVERSE_RELAY_ACTION_MAP[action]}
-                )
-                mock_envoy.update_dry_contact.reset_mock()
-            # and finally back to original
-            await hass.services.async_call(
-                SELECT_DOMAIN,
-                SERVICE_SELECT_OPTION,
-                {
-                    ATTR_ENTITY_ID: test_entity,
-                    ATTR_OPTION: current_state,
-                },
-                blocking=True,
-            )
-            mock_envoy.update_dry_contact.assert_called_once_with(
-                {"id": contact_id, target[2]: REVERSE_RELAY_ACTION_MAP[current_state]}
-            )
-            mock_envoy.update_dry_contact.reset_mock()
+    assert (dry_contact := mock_envoy.data.dry_contact_settings[relay])
+    assert (name := dry_contact.load_name.lower().replace(" ", "_"))
+
+    test_entity = f"{entity_base}{name}_{target[0]}"
+
+    assert (entity_state := hass.states.get(test_entity))
+    assert RELAY_ACTION_MAP[target[1]] == entity_state.state
```

Comment:
Assert expected on the right (`assert actual == expected`):

```suggestion
    assert entity_state.state == RELAY_ACTION_MAP[target[1]]
```